src/cryptonote_core/tx_pool.cpp

```diff
@@ -751,13 +753,9 @@ namespace cryptonote
     m_remove_stuck_tx_interval.do_call([this](){return remove_stuck_transactions();});
   }
   //---------------------------------------------------------------------------------
-  sorted_tx_container::iterator tx_memory_pool::find_tx_in_sorted_container(const crypto::hash& id) const
+  sorted_tx_container::iterator tx_memory_pool::find_tx_in_sorted_container(const crypto::hash& id)
```

Comment:
I looked at the documentation, and it looks like this function can remain `const`? What broke this?

Reply:
I just noticed that `iterator` has to be returned, so presumably the reason is `find` and `project_up` return `const_iterator` here.

Reply:
Yes you are right. If it remains `const` then the `find` will return `const_iterator` and that is causing a lot headache. We want to use that iterator in `erase` which doesn't accept `const_iterator`, and no easy way of turning `const_iterator` to `iterator` in boost.